modules/planning/on_lane_planning.cc

```diff
@@ -745,5 +746,161 @@ void OnLanePlanning::AddPartitionedTrajectory(
   }
 }
 
+void OnLanePlanning::AddStitchSpeedProfile(
+    planning_internal::Debug* debug_chart) {
+  if (!FrameHistory::Instance()->Latest()) {
+    AINFO << "Planning frame is empty!";
+    return;
+  }
+
+  // if open space info provider success run
+  if (!frame_->open_space_info().open_space_provider_success()) {
+    return;
+  }
+
+  auto chart = debug_chart->mutable_planning_data()->add_chart();
+  chart->set_title("Open Space Speed Plan Visualization");
+  auto* options = chart->mutable_options();
+  // options->mutable_x()->set_mid_value(Clock::NowInSeconds());
+  options->mutable_x()->set_window_size(20.0);
+  options->mutable_x()->set_label_string("time (s)");
+  options->mutable_y()->set_min(2.1);
+  options->mutable_y()->set_max(-1.1);
+  options->mutable_y()->set_label_string("speed (m/s)");
+
+  // auto smoothed_trajectory = open_space_debug.smoothed_trajectory();
+  auto* speed_profile = chart->add_line();
+  speed_profile->set_label("Speed Profile");
+  const auto& last_trajectory =
+      FrameHistory::Instance()->Latest()->current_frame_planned_trajectory();
+  for (const auto& point : last_trajectory.trajectory_point()) {
+    auto* point_debug = speed_profile->add_point();
+    point_debug->set_x(point.relative_time() +
+                       last_trajectory.header().timestamp_sec());
+    point_debug->set_y(point.v());
+  }
+  // Set chartJS's dataset properties
+  auto* speed_profile_properties = speed_profile->mutable_properties();
+  (*speed_profile_properties)["borderWidth"] = "2";
+  (*speed_profile_properties)["pointRadius"] = "0";
+  (*speed_profile_properties)["lineTension"] = "0";
+  (*speed_profile_properties)["fill"] = "false";
+  (*speed_profile_properties)["showLine"] = "true";
+}
+
+void OnLanePlanning::AddPublishedSpeed(ADCTrajectory* const trajectory_pb,
+                                       planning_internal::Debug* debug_chart) {
```

Comment:
Accept const reference as input. Don't let others break you with a NULL or wild pointer.

Reply:
Done with other minor cleaning up to avoid confusion.